More about #9718: Updating the docs (#18623)
yiisoft/yii2: docs/guide/security-authentication.md

If a particular method is not needed, you may implement it with an empty body. For example, if your application
is a pure stateless RESTful application, you would only need to implement [[yii\web\IdentityInterface::findIdentityByAccessToken()|findIdentityByAccessToken()]]
and [[yii\web\IdentityInterface::getId()|getId()]] while leaving all other methods with an empty body. Or if your 

> application uses session only authentication, you would need to implement only the methods [[yii\web\IdentityInterface::findIdentity()|findIdentity()]] and 
> [[yii\web\IdentityInterface::getId()|getId()]], while leaving the rest of the methods with an empty body.

Reviewer: ```suggestion
application uses session only authentication, you would need to implement all the methods except
[[yii\web\IdentityInterface::findIdentityByAccessToken()|findIdentityByAccessToken()]].
```

    }

    /**

>      * @return int|string current user ID

Reviewer: It was alright.

     * Get thecurrent user's ID
     * 
     * @return int|string 
     */
    public function getId()
    {
        return $this->id;
    }

    /**

>      * @return string current user auth key

Reviewer: It was alright.

     * @return string|null
     */
    public function getAuthKey()
    {
        return $this->auth_key;
    }

    /**

>      * Checks if the provided auth key is valid for the current user

Reviewer: It was alright.
